### PR TITLE
[BACKLOG-21243] MQTT Consumer - SSL

### PR DESCRIPTION
**Pull request by mkambol**
BaseStreamMeta xml serialization will now use a group tag
if the attribute is part of a injection group.

This will be used to bucket SSL properties under a <SSL> tag.

https://jira.pentaho.com/browse/BACKLOG-21243

### CODE REVIEW

**Comment by mkambol** [comment]
@pentaho/ackbar

File: engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java [hunk 1]
```diff
@@ -75,20 +80,44 @@
     BeanInjectionInfo info = new BeanInjectionInfo( this.getClass() );
     BeanInjector injector = new BeanInjector( info );
     Map<String, BeanInjectionInfo.Property> properties = info.getProperties();
+
+    StringBuilder buider = new StringBuilder();
+
+    info.getGroups()
+      .forEach( group -> {
+        boolean groupHasName = !isNullOrEmpty( group.getName() );
```
Comment (kurtwalker) [inline engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java]: Can we move this lambda to a method and use a method reference?  It's pretty big for inline lambda.  Giving it a name will help understand what it's doing too.

File: engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java [hunk 2]
```diff
@@ -75,20 +80,44 @@
...
+
+          properties.entrySet().stream()
+            .filter( entry -> groupProps.contains( entry.getValue() ) )
+            .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) )
+        ) );
+        if ( groupHasName ) {
+          buider.append( "</" ).append( group.getName() ).append( ">\n" );
+        }
+      } );
+    return buider.toString();
```
Comment (kurtwalker) [inline engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java]: I'm also wondering if a map to string and a Collectors.joining would be simpler than the StringBuilder.

Comment (mkambol) [reply engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java]: I ended up switching to a StringBuilder after needing to conditionally insert a <group> tag.  Including the opening/closing tag was getting messy, and really this whole mess should probably be done with a DocumentBuilder or real serialization library so we're not assembling string xml.  Didn't want to go that far now, though.

File: engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java (hunk 1 above)
Comment (mkambol) [reply engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java]: Good idea.

**Pull request merged**